core/src/test/scala/org/broadinstitute/dsde/rawls/snapshot/SnapshotServiceSpec.scala

```diff
@@ -108,6 +127,92 @@ class SnapshotServiceSpec extends AnyWordSpecLike with Matchers with MockitoSuga
       )
     }
 
+    "create a WSM workspace if one doesn't exist when creating a snapshot reference" in withMinimalTestDatabase { _ =>
+      val mockSamDAO = defaultMockSamDao()
+
+      val mockWorkspaceManagerDAO = defaultMockWorkspaceManagerDao()
+      when(
+        mockWorkspaceManagerDAO.getWorkspace(any[UUID], any[RawlsRequestContext])
+      ).thenAnswer(_ => new ApiException(404, "Workspace does not exist"))
+
+      val mockDataRepoDAO: DataRepoDAO = new MockDataRepoDAO("mockDataRepo")
+
+      val workspace = minimalTestData.workspace
+
+      val snapshotService = SnapshotService.constructor(
+        slickDataSource,
+        mockSamDAO,
+        mockWorkspaceManagerDAO,
+        "fake-terra-data-repo-dev",
+        mockDataRepoDAO
+      )(testContext)
+
+      // call createSnapshot on the service
+      Await.result(
+        snapshotService.createSnapshot(workspace.toWorkspaceName,
+                                       NamedDataRepoSnapshot(DataReferenceName("foo"),
+                                                             Option(DataReferenceDescriptionField("foo")),
+                                                             UUID.randomUUID()
+                                       )
+        ),
+        Duration.Inf
+      )
+
+      // assert that the service checked to see if the workspace exists
+      verify(mockWorkspaceManagerDAO, times(1)).getWorkspace(
+        any[UUID],
+        any[RawlsRequestContext]
+      )
+
+      // assert that the service called WSM's createWorkspace
+      verify(mockWorkspaceManagerDAO, times(1)).createWorkspace(
+        any[UUID],
+        any[RawlsRequestContext]
+      )
+    }
+
+    "not create a WSM workspace if one already exists when creating a snapshot reference" in withMinimalTestDatabase {
+      _ =>
+        val mockSamDAO = defaultMockSamDao()
+
+        val mockWorkspaceManagerDAO = defaultMockWorkspaceManagerDao()
+
+        val mockDataRepoDAO: DataRepoDAO = new MockDataRepoDAO("mockDataRepo")
+
+        val workspace = minimalTestData.workspace
+
+        val snapshotService = SnapshotService.constructor(
+          slickDataSource,
+          mockSamDAO,
+          mockWorkspaceManagerDAO,
+          "fake-terra-data-repo-dev",
+          mockDataRepoDAO
+        )(testContext)
+
+        // call createSnapshot on the service
+        Await.result(
+          snapshotService.createSnapshot(workspace.toWorkspaceName,
+                                         NamedDataRepoSnapshot(DataReferenceName("foo"),
+                                                               Option(DataReferenceDescriptionField("foo")),
+                                                               UUID.randomUUID()
+                                         )
+          ),
+          Duration.Inf
+        )
+
+        // assert that the service checked to see if the workspace exists
+        verify(mockWorkspaceManagerDAO, times(1)).getWorkspace(
+          any[UUID],
```

Comment:
I like using `any` matchers by default when setting up test _stubs_, but I think using `any` in a _verify_ can let a certain category of surprise bugs creep through...this is exactly what bit me when I was trying to make changes to the UI and using overly loose verifies (see [comment](https://github.com/DataBiosphere/terra-ui/pull/4235/commits/5ea60c1a957813e441cc919cef7b099226cff67b#r1340175189)) -- the example I provided is even looser than this, I don't even check the arg list...  but hopefully the point makes sense.

Is there a specific workspace ID from the test fixture you can look for here vs. `any`?

I'm okay with leaving the RawlsRequestContext loose assuming it's not particularly important for the code under test.